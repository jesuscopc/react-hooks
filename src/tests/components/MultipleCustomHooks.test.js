import React from 'react'
import { shallow } from "enzyme";
import { MultipleCustomHooks } from '../../components/03-examples/MultipleCustomHooks'
import { useFetch } from '../../hooks/useFetch';
import { useCouter } from '../../hooks/useCouter';
jest.mock('../../hooks/useFetch');
jest.mock('../../hooks/useCouter');

describe('Sui test multiplecustom hooks', () => {
  useCouter.mockReturnValue(() => {
    counter: 10
    increment: () => {}
  });

  test('should be render correctly', () => {
    useFetch.mockReturnValue({
      data: null,
      loading: true,
      error: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect(wrapper).toMatchSnapshot();  
  });

  test('should be show info', () => {
    useFetch.mockReturnValue({
      data: [
        {
          author: 'Jesus',
          quote: 'Hola mundo'
        }
      ],
      loading: false,
      error: null
    });
    const wrapper = shallow(<MultipleCustomHooks />);
    expect( wrapper.find('.alert').exists() ).toBe( false );
    expect( wrapper.find('footer').text().trim() ).toBe( 'Jesus' );
  });
  
});
