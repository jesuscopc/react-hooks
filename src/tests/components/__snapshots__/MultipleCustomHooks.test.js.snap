// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sui test multiplecustom hooks should be render correctly 1`] = `
<div>
  <h1>
    BreakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
  >
    Siguiente quote
  </button>
</div>
`;
